pkg/monitoring/mimir/service.go

```diff
@@ -0,0 +1,161 @@
+package mimir
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/observability-operator/pkg/common/password"
+	"github.com/giantswarm/observability-operator/pkg/common/secret"
+	"github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent"
+)
+
+const (
+	authSecretName               = "mimir-gateway-ingress"
+	authSecretNamespace          = "mimir"
+	mimirSpecificSecretName      = "mimir-basic-auth"
+	mimirSpecificSecretNamespace = "mimir"
+)
+
+type MimirService struct {
+	client.Client
+	PasswordManager password.Manager
+	SecretManager   secret.Manager
+}
+
+func (ms *MimirService) ConfigureMimir(ctx context.Context, mc string) error {
+	logger := log.FromContext(ctx).WithValues("cluster", mc)
+	logger.Info("ensuring mimir config")
+
+	err := ms.CreateAuthSecret(ctx, logger, mc)
+	if err != nil {
+		logger.Error(err, "failed to create mimit auth secret")
+		return errors.WithStack(err)
+	}
+
+	err = ms.CreateIngressSecret(ctx, mc, logger)
+	if err != nil {
+		logger.Error(err, "failed to create mimir ingress secret")
+		return errors.WithStack(err)
+	}
+
+	logger.Info("ensured mimir config")
+
+	return nil
+}
+
+func (ms *MimirService) CreateAuthSecret(ctx context.Context, logger logr.Logger, mc string) error {
+	objectKey := client.ObjectKey{
+		Name:      mimirSpecificSecretName,
+		Namespace: mimirSpecificSecretNamespace,
+	}
+
+	current := &corev1.Secret{}
+	err := ms.Client.Get(ctx, objectKey, current)
+	if apierrors.IsNotFound(err) {
+		logger.Info("Building auth secret")
+
+		password, err := ms.PasswordManager.GeneratePassword(32)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		secretdata := mc + ":" + password
```

Comment:
Do you need to use the username here? Can secretData not be the password only?

Reply:
You're right, no need for username. Changing this